carma_nav2_port_drayage_demo/src/port_drayage_demo.cpp

```diff
@@ -86,14 +134,66 @@ auto PortDrayageDemo::on_mobility_operation_received(
     pose.header.frame_id = "map";
     pose.pose.position.x = longitude;
     pose.pose.position.y = latitude;
+    current_operation_ = operation;
+    current_strategy_params_ = msg.strategy_params;
 
     goal.poses.push_back(std::move(pose));
 
-    follow_waypoints_client_->async_send_goal(goal);
+    auto send_goal_options = rclcpp_action::Client<nav2_msgs::action::FollowWaypoints>::SendGoalOptions();
+    send_goal_options.result_callback =
+      std::bind(&PortDrayageDemo::on_result_received, this, std::placeholders::_1);
+    follow_waypoints_client_->async_send_goal(goal, send_goal_options);
   } catch (const nlohmann::json::exception & e) {
     RCLCPP_ERROR_STREAM(
       get_logger(), "Could not process MobilityOperation message: JSON error: " << e.what());
   }
 }
 
+auto PortDrayageDemo::on_result_received(
+  const rclcpp_action::ClientGoalHandle<nav2_msgs::action::FollowWaypoints>::WrappedResult & result) -> void
+{
+  switch (result.code)
+  {
+    case rclcpp_action::ResultCode::SUCCEEDED:
+    {
+      RCLCPP_INFO(get_logger(), "Goal successfully reached");
```

Comment:
Could you also populate the `m_header` field here similar to https://github.com/usdot-fhwa-stol/carma-platform/blob/70998021ced231d85fe5dd137906660346bbeb06/port_drayage_plugin/src/port_drayage_worker.cpp#L152

Reply:
Updated